src/modules/rlm_files/rlm_files.c

```diff
@@ -212,10 +220,13 @@ static int getusersfile(TALLOC_CTX *ctx, char const *filename, rbtree_t **ptree)
 		next = entry->next;
 		entry->next = NULL;
 
+		char *str;
+		fr_value_box_aprint(NULL, &str, entry->name, NULL);
+
 		/*
 		 *	DEFAULT entries get their own list.
 		 */
-		if (strcmp(entry->name, "DEFAULT") == 0) {
+		if (strcmp(str, "DEFAULT") == 0) {
```

Comment:
you can check for `entry->name->type  == FR_TYPE_STRING` and then just do the `strcmp()`